### DISCUSSION TITLE
DEVICEIP

### DISCUSSION THREAD

**Discussion by kstelios** [jasonacox/tinytuya#201] [General]
Any idea to bypass DEVICEIP changes (unknown ip) because of DHCP server?
Example code 
```

import tinytuya

d = tinytuya.OutletDevice('DEVICEID', 'DEVICEIP', 'DEVICEKEY')
d.set_version(3.3)
data = d.set_status(True)
```

**Comment by jasonacox** [comment]
Hi @kstelios  - Yes!  You can have tinytuya scan for the device and discover its IP automatically by using `'Auto'` or `None` as the DEVICEIP.  Note: As you would probably expect, this causes a delay while TinyTuya scans for the device.

```python
import tinytuya

d = tinytuya.OutletDevice('DEVICEID', 'Auto', 'DEVICEKEY')
# d.set_version(3.3) - Not needed since scan will determine version too
print(d.status())
```

[ Also, make sure you are running the latest version of tinytuya (1.7.1) as there has been significant improvements to the discovery logic thanks to @uzlonewolf ]
